src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -300,76 +273,72 @@ static ucs_status_t uct_rdmacm_cm_create_dummy_qp(struct rdma_cm_id *id,
     return UCS_OK;
 }
 
-static ucs_status_t uct_rdamcm_cm_ep_create_qpn(uct_rdmacm_cm_ep_t *cep,
-                                                struct uct_rdmacm_cm_qp_context *ctx)
+static uint32_t
+uct_rdamcm_cm_ep_create_qpn(uct_rdmacm_cm_ep_t *cep,
+                            struct uct_rdmacm_cm_device_context *ctx)
 {
     ucs_status_t status;
 
     if (ctx->reserved_qpn_support) {
         status = uct_rdamcm_cm_ep_create_reserved_qpn(cep, ctx);      
     } else {
         /* create a dummy qp in order to get a unique qp_num to provide to librdmacm */
-        status = uct_rdmacm_cm_create_dummy_qp(cep->id, ctx->dummy_qp_ctx.cq,
-                                               &cep->qp);
+        status = uct_rdmacm_cm_create_dummy_qp(cep->id, ctx->cq, &cep->qp);
     }
+    ucs_assert_always(status == UCS_OK);
 
-    if (status == UCS_OK) {
-        cep->flags |= UCT_RDMACM_CM_EP_QPN_CREATED;
-    }
+    cep->flags |= UCT_RDMACM_CM_EP_QPN_CREATED;
 
-    return status;
+    if (ctx->reserved_qpn_support) {
+        return cep->qpn;
+    } else {
+        return cep->qp->qp_num;
+    }
 }
 
 static void uct_rdamcm_cm_ep_destroy_qpn(uct_rdmacm_cm_ep_t *cep)
 {
-    struct uct_rdmacm_cm_qp_context *ctx;
+    struct uct_rdmacm_cm_device_context *ctx;
     ucs_status_t status;
 
     if (!(cep->flags & UCT_RDMACM_CM_EP_QPN_CREATED)) {
         return;
     }
 
-    status = uct_rdmacm_cm_get_qp_context(uct_rdmacm_cm_ep_get_cm(cep),
-                                          cep->id->verbs, &ctx);
+    status = uct_rdmacm_cm_get_device_context(uct_rdmacm_cm_ep_get_cm(cep),
+                                              cep->id->verbs, &ctx);
     if (status != UCS_OK) {
-        ucs_error("failed to get qp context");
+        ucs_error("failed to get device context");
```

Comment:
pls print more info in this error, or remote it altogether since uct_rdmacm_cm_get_device_context() already prints an error